zephyr/include/sof/lib/regions_mm.h

```diff
@@ -10,31 +10,163 @@
 
 #include <adsp_memory.h>
 #include <adsp_memory_regions.h>
+#include <ipc/topology.h>
+#include <rtos/alloc.h>
 #include <zephyr/drivers/mm/system_mm.h>
 #include <zephyr/sys/mem_blocks.h>
 #include <zephyr/init.h>
-#include <ipc/topology.h>
+#include <sof/list.h>
 
-/*
- * Struct containing information on virtual memory heap.
- * Information about allocated physical blocks is stored in
- * physical_blocks_allocators variable and uses zephyr memblocks api.
+/* Dependency on ipc/topology.h created due to memory capability definitions
+ * that are defined there
+ */
+
+/* Defines maximum amount of memory block allocators in one heap
+ * since minimum realistic block should be DCASHE_LINE size
+ * and block sizes in allocator must be powers of 2
+ * so logicly it gives limited block sizes
+ * eg allocator of block size 64 128... 1024 8192
+ * would end up in 8 allocators.
+ * It is expected that allocations bigger than that would
+ * either be spaned on specifically configured heap
+ */
+#define MAXIMUM_BLOCK_SIZES 8
+
+/* list of virtual_memory_heap objects created */
+struct list_item vm_heaps;
+/* list of vmh_active_physical_mapping objects created */
+struct list_item physical_mappings;
+
+/** @struct virtual_memory_heap
+ *
+ *  @brief This structure holds all information about virtual memory heap
+ *  it aggregates information about its allocations and
+ *  physical mappings.
+ *
+ *  @var node generic list marker
+ *  @var virtual_region pointer to virtual region information, it holds its
+ *  attributes size and beginning ptr provided by zephyr.
+ *  @var physical_blocks_allocators[] a table of block allocators
+ *  each representing a virtual regions part in blocks of a given size
+ *  governed by sys_mem_blocks api.
+ *  @var active_allocations list of current heap allocations across
+ *  all its allocators (sys_mem_blocks) saved in vmh_active_physical_mapping
+ *  struct.
+ *  TO DO: mem_blocks api should be broadened to allow saving
+ *  allocations to a bitarray.
+ *  @var span configuration value deciding if heap allocations
+ *  will be contiguous or single block.
  */
 struct virtual_memory_heap {
```

Comment:
I think `struct heap_config *cfg` should be stored here too.
What if I during runtime consider reconfiguration of the heap based on its current form?
Consider following scenario:
1. I've initialized and got virtual_memory_heap *my_awesome_heap .
2. I pass this heap to someAPI1 and someAPI2. Runtime conditions of the system may allocate memory to the heap
in someAPI1 or alternatively, reconfigure my_awesome_heap and then allocate something there.
3. I've exited logic of someAPI1 and someAPI2 and I would like to allocate something else on my_awesome_heap but only if it maintains a form I gave it during first initialization. To get this info I would need to analyze heap_config right?